flink-connectors/flink-hbase/src/main/java/org/apache/flink/addons/hbase/HBaseTableSourceInputFormat.java

```diff
@@ -0,0 +1,160 @@
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.types.Row;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.TableNotFoundException;
+import org.apache.hadoop.hbase.client.HTable;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.ConnectionFactory;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.hbase.util.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link InputFormat} subclass that wraps the access for HTables. Returns the result as {@link Row}
+ */
+public class HBaseTableSourceInputFormat extends TableInputFormat<Row> implements ResultTypeQueryable<Row> {
+
+	private static final long serialVersionUID = 1L;
+
+	private static final Logger LOG = LoggerFactory.getLogger(HBaseTableSourceInputFormat.class);
+	private String tableName;
+	private transient Connection conn;
+	private transient org.apache.hadoop.conf.Configuration conf;
+	private HBaseTableSchema schema;
+
+	public HBaseTableSourceInputFormat(org.apache.hadoop.conf.Configuration conf, String tableName, HBaseTableSchema schema) {
+		this.tableName = tableName;
+		this.conf = conf;
+		this.schema = schema;
+	}
+
+	@Override
+	public void configure(Configuration parameters) {
+		LOG.info("Initializing HBaseConfiguration");
+		connectToTable();
+		if(table != null) {
+			scan = getScanner();
+		}
+	}
+
+	@Override
+	protected Scan getScanner() {
+		// TODO : Pass 'rowkey'. For this we need FilterableTableSource
+		Scan scan = new Scan();
+		Map<String, List<Pair>> familyMap = schema.getFamilyMap();
+		for(String family : familyMap.keySet()) {
+			// select only the fields in the 'selectedFields'
+			List<Pair> colDetails = familyMap.get(family);
+			for(Pair<String, TypeInformation<?>> pair : colDetails) {
+				scan.addColumn(Bytes.toBytes(family), Bytes.toBytes(pair.getFirst()));
+			}
+		}
+		return scan;
+	}
+
+	@Override
+	public String getTableName() {
+		return tableName;
+	}
+
+	@Override
+	protected Row mapResultToTuple(Result res) {
+		List<Object> values = new ArrayList<Object>();
+		int i = 0;
+		Map<String, List<Pair>> familyMap = schema.getFamilyMap();
+		Row[] rows = new Row[familyMap.size()];
+		for(String family : familyMap.keySet()) {
+			List<Pair> colDetails = familyMap.get(family);
+			for(Pair<String, TypeInformation<?>> pair : colDetails) {
+				byte[] value = res.getValue(Bytes.toBytes(family), Bytes.toBytes(pair.getFirst()));
+				if(value != null) {
+					values.add(schema.deserialize(value, pair.getSecond()));
+				} else {
+					values.add(schema.deserializeNull(pair.getSecond()));
```

Comment:
Ya. I dint know this. Your question answers my doubt and the shaky method that I added can be removed.